### ISSUE TITLE
Perplexity in T5/BART

### ISSUE THREAD

**Issue by Palipoor** [huggingface/transformers#4210]
Hi,
Is there a way to compute perplexity for T5 or BART outputs?

**Comment by patil-suraj** [comment]
T5 uses CrossEntropy loss so I think you can use `torch.exp(loss)` to calculate perplexity.

**Comment by Palipoor** [comment]
Thanks!

**Comment by Palipoor** [comment]
@patil-suraj Sorry to bother you again, I thought my problem has been solved, but I don't know how to deal with the loss. It's a (1,1,vocab_size) tensor. I don't even have an intuition about what that means. Can you help me?

**Comment by patil-suraj** [comment]
It seems that you are not passing `lm_labels`. When you pass `lm_labels` to T5/BART's forward methods then the first element of the output tuple is the loss, otherwise its `prediction_scores` tensor which you are receiving.

**Comment by fajemila** [comment]
hi @patil-suraj bart doesnt have lm_labels has argument

**Comment by patil-suraj** [comment]
`lm_labels` is renamed  to `labels`

**Comment by fajemila** [comment]
Thanks for the prompt response, @patil-suraj , previously before trying Bart, I trained it with t5,the valid loss gets to almost zero, but still found it difficult generating correct translation. I would love to know reasons for that. Thanks

**Comment by patil-suraj** [comment]
I'm not sure which problem you are talking about, hard to answer without knowing the context.

**Comment by fajemila** [comment]
Hi patil,

I am working on a translation task/competition, so as a text2text
framework, I used t5 even though it is not pretrained on my source language
just for the knowledge sake,
The notebook link here

https://colab.research.google.com/drive/1w0brh9KULV4zDmgMme4csXeCpfAR960N?usp=sharing

the test and valid loss was declining to zero even though the later
generated predictions wasn't good so I changed to Bart even though no
improvement, I have no idea why? I would love to know the reason.

I found MarianMT just yesterday which was pretrained for my task yo-en, I
have started working with it. I think it would also be the case for MarianMt
Thank you.
